aml-development/ozp-backend
# refactor(delete category): identify orphans

**Not Ready. New Requirements**

AMLNG-XXX

**Description**     
*Copy and paste Description from orginal issue*  

**Acceptance Criteria**   
*Copy and paste Acceptance Criteria from or-ginal issue*
 
**How to test code**    
1. You need *both* back-end and front-end 'delete_category' branches   
            -Checkout back-end 'delete_category' branch
            -Checkout front-end 'delete_category' branch

2. _Test: Category with no associated listings_
-    From  'Center Home'  ensure that your category is has no listings
-    =Hamburger menu --> Manage: Center Settings
-    Select the category to be deleted
-    Press the 'Delete' selected record button(trash can)
-    Press 'Delete' button, in the Message box "Are you sure ...?"
 Expected Result: The Category will be deleted

3. _Test: Category with Listings that have multiple categories, and  listings that only include the category you're deleting._
-     (Use 'Center Home' edit listing  to setup categories for your test cases)
-    =Hamburger menu --> Manage: Center Settings
-    Select the category to be deleted
-    Press the 'Delete' selected record button(trash can)
-    Press 'Delete' button, in the Message box "Are you sure ...?"
Expected Result: Message box displayed: Cannot delete category. Displays titles and number of listings that must be moved to another category before a deletion can be completed.

4. _Test: Category with Listings that have multiple categories. (No orphans)
-     (Use 'Center Home' edit listing  to setup categories for your test cases)
-    =Hamburger menu --> Manage: Center Settings
-    Select the category to be deleted
-    Press the 'Delete' selected record button(trash can)
-    Press 'Delete' button, in the Message box "Are you sure ...?"
Expected Result:  Category is deleted

**Please Review**     
@aml-development/ozp-backend-team    

**Checklist**
